### ISSUE TITLE
not handling objects inside arrays for indent rule

### ISSUE THREAD

**Issue by piotr-s-brainhub** [eslint/eslint#9720]
* **ESLint Version:** 3.19.0
* **Node Version:** 8.6.0
* **npm Version:** 5.2.0
**What parser (default, Babel-ESLint, etc.) are you using?** default

**General description:**
`indent` rule does not handle objects inside arrays. On the other hand it works fine with variables, strings or numbers as in examples on https://eslint.org/docs/rules/indent .

**Please show your full configuration:**
```json
{
  "extends": "airbnb-base",
  "rules": {
    "indent": ["error", 2, { "ArrayExpression": 1, "ObjectExpression": 1 }]
  }
}
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
module.exports = [
        {},
    {},
        {},
];
```

```bash
./node_modules/.bin/eslint foo.js
```

**What did you expect to happen?**
It should throw an error.
**What actually happened? Please include the actual, raw output from ESLint.**
It shows an empty output.

**Comment by gyandeeps** [comment]
Have tried upgrading to 4.x as 4.x release has a new indent (rewritten) rule?

**Comment by piotr-s-brainhub** [comment]
It works fine for eslint 4 (exactly 4.13.1). Thanks
